addons/website/views/website_navbar_templates.xml

```diff
@@ -77,8 +77,18 @@
                         <a data-action="show-mobile-preview" href="#"><span title="Mobile preview" role="img" aria-label="Mobile preview" class="fa fa-mobile"/></a>
                     </li>
                     <li t-if="user_id.has_group('base.group_multi_company') and user_id.has_group('website.group_multi_website') and len(multi_website_companies)">
-                        <a class="dropdown-toggle waves" data-toggle="dropdown" href="#"><t t-esc="multi_website_companies_current['name']"/></a>
+                        <a class="dropdown-toggle waves" data-toggle="dropdown" href="#">
+                            <span class="d-block d-lg-none">
+                                <i class="fa fa-building"></i>
```

Comment:
Do not wrap the icon in a span, just pust the d-* classes on the icon. Also, d-block is probably useless, d-lg-none should be enough